File: src/main/java/mnm/mods/tabbychat/client/UserChannel.java
package mnm.mods.tabbychat.client;

public class UserChannel extends AbstractChannel {

    private String name;

    UserChannel(String name) {
        super(name);
        this.name = name;
        setPrefix("/msg " + name);
    }

    @Override
    public String getName() {
        return name;
    }

    @Override
    public String getDisplayName() {
        return toString();
    }

    @Override
    public int hashCode() {
        return toString().hashCode();
    }

    @Override
    public String toString() {
        return "@" + name;
    }
}

File: src/main/kotlin/mnm/mods/tabbychat/client/UserChannel.kt
package mnm.mods.tabbychat.client

class UserChannel internal constructor(override val name: String) : AbstractChannel(name) {

    override val displayName: String = "@$name"

    init {
        prefix = "/msg $name"
    }

    override fun hashCode(): Int {
        return toString().hashCode()
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as UserChannel

        if (name != other.name) return false

        return true
    }

    override fun toString(): String {
        return displayName
    }

}
